### PR TITLE
Add Final fees to Liquidatable

### LINKED ISSUE #876: Add final fee payment to the Expiring Multiparty contract

### PR DESCRIPTION
**Pull request by nicholaspai**
Resolves #876 

Requires `disputer` to pay the final fee (in addition to the `disputeBond`) when calling `dispute()` which makes an Oracle Request

As I've currently set it up, there is no need to modify `feeAdjustment` that Matt added in #869 since I just require the `msg.sender` of `dispute()` to pay the final fee immediately (similar to how we handle the `disputeBond`)

### CODE REVIEW

File: core/contracts/financial_templates/Liquidatable.sol [hunk 1]
```diff
@@ -232,6 +232,15 @@ contract Liquidatable is PricelessPositionManager {
...
 
+        // Pay a final fee
+        StoreInterface store = StoreInterface(finder.getImplementationAddress("Store"));
+        FixedPoint.Unsigned memory finalFee = store.computeFinalFee(address(collateralCurrency));
+        if (finalFee.isGreaterThan(0)) {
+            require(collateralCurrency.transferFrom(msg.sender, address(this), finalFee.rawValue));
+            collateralCurrency.safeIncreaseAllowance(address(store), finalFee.rawValue);
+            store.payOracleFeesErc20(address(collateralCurrency));
+        }
+
```
Comment (mrice32) [inline core/contracts/financial_templates/Liquidatable.sol]: WDYT about reusing the payFinalFees() in the FeePayer contract (introduced in #915) to do this operation? To make it work, you may have to add a parameter like `address from` so the method knows who, if anyone to pull the fee from.

Comment (nicholaspai) [reply core/contracts/financial_templates/Liquidatable.sol]: The main consideration is that I want the `msg.sender` to transfer the `finalFee` before paying fees. This way the contract definitely has enough funds to send the final fee. 

I could check if the address passed in is either `address(0)` (or `address(this)`), then no need to call the `transferFrom` line

Comment (mrice32) [reply core/contracts/financial_templates/Liquidatable.sol]: Agreed, you'd have to do the check so you don't end up doing a transferFrom on yourself.

Comment (nicholaspai) [reply core/contracts/financial_templates/Liquidatable.sol]: yeah i think that makes sense ill add

**Comment by mrice32** [comment]
@nicholaspai I'm gonna wait to review this until we get #915 in since they include some of the same code.

**Pull request merged**